src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt

```diff
@@ -41,7 +42,7 @@ private fun checkFunction(holder: AnnotationHolder, fn: RsFunction) {
         is RsFunctionOwner.Trait -> {
             deny(fn.default, holder, "${fn.title} cannot have the `default` qualifier")
             deny(fn.vis, holder, "${fn.title} cannot have the `pub` qualifier")
-            deny(fn.const, holder, "Trait functions cannot be declared const [E0379]")
+            if (fn.const != null) RsDiagnostic.ConstTraitFnError(fn.const!!).addToHolder(holder)
```

Comment:
This can be replaced with `fn.const?.let { RsDiagnostic.ConstTraitFnError(it).addToHolder(holder) }` or just unwrapped to a local variable. Using of `!!` is a very bad practice (but it used widely in this file, I know)